### DISCUSSION TITLE
Support for multi Gemfiles with a single lock file

### DISCUSSION THREAD

**Discussion by jorge-wonolo** [renovatebot/renovate#22710] [Suggest an Idea]
### Tell us more.

Ruby allows for multiple Gemfiles to be imported/added into a single project's Gemfile at runtime.  You can see a sample of this in https://github.com/jorge-wonolo/renovate-test

Having multiple gemfiles in a projects provides multiple benefits from ownership, packaging of dependencies and better separation of gems in a monolith or monorepo.

Currently the bundler setup expected a matching gemfile.lock when analyzing the gems on a repository https://github.com/renovatebot/renovate/blob/b7ae617eb66232d197f728c908163578b22465fc/lib/modules/manager/bundler/extract.ts#L213 therefore when using a `update-lockfile` strategy it fails to identify the proper matching lock file and failing to suggest updates.

Ideally renovate should be able to use the single lock file for all subsequent gemfile processing and/or allow for a single lock file to be specified in the config.

**Comment by HonkingGoose** [comment]
Follow issue:

- #22725
